docs/how_to_sign_commits_using_the_gitpython_package.md

```diff
@@ -0,0 +1,404 @@
+# How to sign commits using the GitPython package
+
+> [GitPython](https://github.com/gitpython-developers/GitPython) is a python library used to interact with git repositories, high-level like git-porcelain, or low-level like git-plumbing.
+
+GitPython does not allow you to sign commits with its `git-porcelain` method. The way you can committ with GitPython is:
+
+```python
+repo = Repo(repo_dir)
+index = repo.index
+index.add([file_to_commit_path])
+author = Actor("An author", "author@example.com")
+
+index.commit("my commit message", author=author)
+```
+
+`index.commit` method does not provide any argument to sign commits. If you want to sign commits you have to use the the `git-plumbing` method `git.commit(...)`:
+
+```python
+repo = Repo(repo_dir)
+index = repo.index
+index.add([file_to_commit_path])
+author = Actor("An author", "author@example.com")
+
+index.write()
+repo.git.commit('-S', f'--gpg-sign={signingkey}', '-m', "my commit message")
+```
+
+The `repo.git.commit` method is basically a wrapper for the git commit command. All the arguments you pass to it will be passed to the git commit command. Notice you have to call two methods:
+
+```python
+index.write()
+repo.git.commit(arg, ...)
+```
+
+I do not not exactly why you need to `write` your changes but it something the git commit wrapper does:
+
+<https://github.com/gitpython-developers/GitPython/blob/88732b694068704cb151e0c4256a8e8d1adaff38/git/index/base.py#L938-L957>
+
+And it's something [Sebastian Thiel](https://github.com/gitpython-developers/GitPython/issues/580#issuecomment-282474086) said it was needed.
+
+So if you want to sign the commit you only have to specify the signing key. The GitPython package relies on Git configuration and Git relies on GPG configuration. What is happending under the hood is:
+
+1. GitPython is going to call the git commit command using a [git binary wrapper](https://github.com/gitpython-developers/GitPython/blob/254305c935893d7578b112acfa814a07d398ae28/git/cmd.py#L171).
+2. Git is going to call GPG command to sign the commit.
+3. GPG is going to prompt you with the passphrase of the key.
+
+As long as the GPG and Git configuration are OK, GitPython will sign the commit correctly. The [example 03](../03_sign_commit_using_the_gitpython_package) in this repo creates a new signed commit, but before signing, it also sets up the GPG and Git configuration needed. I'm not going to explain how you can create a GPG key and sign commits manually becuase you can follow other tutorials like:
```

Comment:
- Minor typo: "because"
- "I'm not going to explain you...". A little rude, isn't it? Maybe better "In case you want to know how you can..."